src/frontend/src/session.rs

```diff
@@ -249,18 +251,41 @@ impl FrontendEnv {
 pub struct SessionImpl {
     env: FrontendEnv,
     database: String,
+    /// Stores the value of configurations.
+    config_map: RwLock<HashMap<String, ConfigEntry>>,
+}
+
+#[derive(Clone)]
+pub struct ConfigEntry {
```

Comment:
It's better to be an enum.

Reply:
Think twice and I still suggest not do this now. If we change to `enum` or introduce `registry`, it might be hard to design a good API (I guess still have to use generics). There maybe a lot of efforts on this PR. For now, we'd better not bothering too much on user's invalid input and redcue the complexity.

Reply:
Got it.